Metalama.Framework.Engine/Templating/SyntaxTreeAnnotationMap.cs

```diff
@@ -325,24 +325,19 @@ public IEnumerable<ISymbol> GetCandidateSymbols( SyntaxNode node )
                 case ArgumentSyntax argument:
                     var invocation = node.FirstAncestorOrSelf<InvocationExpressionSyntax>();
 
-                    if ( invocation != null )
+                    if ( invocation != null && this.GetSymbol( invocation.Expression ) is IMethodSymbol invokedMethod )
```

Comment:
Consider also the cases where the invoked expression is not a method, i.e. it can be an event or a delegate. Add tests for these cases.

Reply:
I have added tests that call this method with event and delegate invocation.

But they don't actually exercise this part of the code. In fact, I don't know why this code exists: removing the whole `case` doesn't fail any tests.

Reply:
Ok so let's remove the whose `case`. Anyway it's probably incorrect because the type of an argument is not necessarily the one of its parameters, because it does not handle `params` and so on.